src/sentry/testutils/perfomance_issues/store_transaction.py

```diff
@@ -0,0 +1,81 @@
+import random
+from datetime import datetime, timedelta
+from typing import Sequence
+from unittest import mock
+
+from django.utils import timezone
+
+from sentry.models import Environment, Group
+from sentry.testutils import SnubaTestCase
+
+
+class PerfIssueTransactionTestMixin:
+    def store_transaction(
+        self: SnubaTestCase,
+        project_id: int,
+        user_id: str,
+        groups: Sequence[Group],
+        environment: Environment = None,
+        timestamp: datetime = None,
+    ):
+        from sentry.event_manager import _pull_out_data
+        from sentry.utils import snuba
+
+        # truncate microseconds since there's some loss in precision
+        insert_time = (timestamp if timestamp else timezone.now()).replace(microsecond=0)
+
+        user_id_val = f"id:{user_id}"
+
+        def inject_group_ids(jobs, projects):
+            _pull_out_data(jobs, projects)
+            for job in jobs:
+                job["event"].groups = groups
+            return jobs, projects
+
+        # XXX: bypass having to fake performance issue by intentionally assigning this transaction to a group
+        with mock.patch("sentry.event_manager._pull_out_data", inject_group_ids):
+            event_data = {
+                "type": "transaction",
+                "level": "info",
+                "message": "transaction message",
+                "tags": [("sentry:user", user_id_val)],
+                "contexts": {"trace": {"trace_id": "b" * 32, "span_id": "c" * 16, "op": ""}},
+                "timestamp": insert_time.timestamp(),
+                "start_timestamp": insert_time.timestamp(),
+                "received": insert_time.timestamp(),
+                #
```

Comment:
Leftover `#` - can this be used to replace all the instances of `hack_pull_out_data` we've been using for perf tests? I know there are more like `tests/sentry/eventstore/snuba/test_backend.py` for example. Might also be good to check in w/ @udameli since she's working on ISP-16.

Reply:
Good call out. I know we have the `hack_pull_out_data` stuff spread throughout the code which, hopefully, would get cleaned up after @udameli gets her PR in. I think it makes sense to centralize all that when that gets merged in.